packages/manager/src/features/CloudPulse/shared/TimeRangeSelect.tsx

```diff
@@ -0,0 +1,143 @@
+import * as React from 'react';
+
+import Select, {
+  BaseSelectProps,
+  Item,
+} from 'src/components/EnhancedSelect/Select';
+
+interface Props
+  extends Omit<
+    BaseSelectProps<Item<Labels, Labels>, false>,
+    'defaultValue' | 'onChange'
+  > {
+  defaultValue?: Labels;
+  handleStatsChange?: (start: number, end: number) => void;
+}
+
+const _PAST_7_DAYS = 'Past 7 Days';
+const _PAST_12_HOURS = 'Past 12 Hours';
+const _PAST_24_HOURS = 'Past 24 Hours';
+const _PAST_30_DAYS = 'Past 30 Days';
+const _PAST_30_MINUTES = 'Past 30 Minutes';
```

Comment:
Our general SNAKE_CASE convention includes underscores only between words
```suggestion
const PAST_7_DAYS = 'Past 7 Days';
const PAST_12_HOURS = 'Past 12 Hours';
const PAST_24_HOURS = 'Past 24 Hours';
const PAST_30_DAYS = 'Past 30 Days';
const PAST_30_MINUTES = 'Past 30 Minutes';
```